src/api.js

```diff
@@ -0,0 +1,797 @@
+    ptr = sqlite3_column_blob(this.stmt, pos);
+    result = new Uint8Array(size);
+    for (i = k = 0, ref = size; 0 <= ref ? k < ref : k > ref; i = 0 <= ref ? ++k : --k) {
+        result[i] = HEAP8[ptr + i];
+    }
+    return result;
+};
+
+/* Get one row of results of a statement.
+If the first parameter is not provided, step must have been called before get.
+@param [Array,Object] Optional: If set, the values will be bound to the statement, and it will be executed
+@return [Array<String,Number,Uint8Array,null>] One row of result
+
+@example Print all the rows of the table test to the console
+
+    var stmt = db.prepare("SELECT * FROM test");
+    while (stmt.step()) console.log(stmt.get());
+ */
+Statement.prototype["get"] = function(params) {
+    var field, k, ref, results1;
+    if (params != null) {
+        this["bind"](params) && this["step"]();
+    }
+    results1 = [];
+    for (field = k = 0, ref = sqlite3_data_count(this.stmt); 0 <= ref ? k < ref : k > ref; field = 0 <= ref ? ++k : --k) {
+        switch (sqlite3_column_type(this.stmt, field)) {
+        case SQLITE_INTEGER:
+        case SQLITE_FLOAT:
+            results1.push(this.getNumber(field));
+            break;
+        case SQLITE_TEXT:
+            results1.push(this.getString(field));
+            break;
+        case SQLITE_BLOB:
+            results1.push(this.getBlob(field));
+            break;
+        default:
+            results1.push(null);
+        }
+    }
+    return results1;
+};
+
+/* Get the list of column names of a row of result of a statement.
+@return [Array<String>] The names of the columns
+@example
+
+    var stmt = db.prepare("SELECT 5 AS nbr, x'616200' AS data, NULL AS null_value;");
+    stmt.step(); // Execute the statement
+    console.log(stmt.getColumnNames()); // Will print ['nbr','data','null_value']
+ */
+Statement.prototype["getColumnNames"] = function() {
+    var i, k, ref, results1;
+    results1 = [];
+    for (i = k = 0, ref = sqlite3_data_count(this.stmt); 0 <= ref ? k < ref : k > ref; i = 0 <= ref ? ++k : --k) {
+        results1.push(sqlite3_column_name(this.stmt, i));
+    }
+    return results1;
+};
+
+/* Get one row of result as a javascript object, associating column names with
+their value in the current row.
+@param [Array,Object] Optional: If set, the values will be bound to the statement, and it will be executed
+@return [Object] The row of result
+@see [Statement.get](#get-dynamic)
+
+@example
+
+    var stmt = db.prepare("SELECT 5 AS nbr, x'616200' AS data, NULL AS null_value;");
+    stmt.step(); // Execute the statement
+    console.log(stmt.getAsObject()); // Will print {nbr:5, data: Uint8Array([1,2,3]), null_value:null}
+ */
+Statement.prototype["getAsObject"] = function(params) {
+    var i, k, len, name, names, rowObject, values;
+    values = this["get"](params);
+    names = this["getColumnNames"]();
+    rowObject = {};
+    for (i = k = 0, len = names.length; k < len; i = ++k) {
```

Comment:
we don't need two variables

Reply:
implemented suggestion in commit 242813f16f2f199d2faf6168879bcf74e7b406e8